torch/distributed/pipeline/sync/microbatch.py

```diff
@@ -132,39 +141,44 @@ def _setitem_by_slice(self, index: slice, value: Tensors) -> None:
         self.value = value[0]
 
 
-def check(input: TensorOrTensors) -> None:
+def check(*inputs) -> None:
     """Checks whether the input is a tensor or tensors.
 
     Raises:
         TypeError: input is not a tensor or tensors.
 
     """
-    if isinstance(input, Sequence):
-        for x in input:
-            if not isinstance(x, Tensor):
-                raise TypeError(f"expected Tensor, but got {input.__class__.__name__}")
-        return
+    for input in inputs:
+        if isinstance(input, Sequence) and len(inputs) == 1:
+            for x in input:
+                if not isinstance(x, Tensor):
+                    raise TypeError(f"expected Tensor, but got {input.__class__.__name__}")
+            return
 
-    if not isinstance(input, Tensor):
-        raise TypeError(f"expected Tensor, but got {input.__class__.__name__}")
+        if not isinstance(input, Tensor):
+            raise TypeError(f"expected Tensor, but got {input.__class__.__name__}")
 
 
-def scatter(input: TensorOrTensors, chunks: int) -> List[Batch]:
+def scatter(*inputs, chunks: int) -> List[Batch]:
     """Splits an input mini-batch into multiple micro-batches."""
-    inputs: Iterable[TensorOrTensors]
-
-    if isinstance(input, Tensor):
-        inputs = input.chunk(chunks)
+    is_single_sequence = False
+    if len(inputs) == 1 and isinstance(inputs[0], Tensor):
+        unwrapped_inputs = inputs[0].chunk(chunks)
     else:
         rotated: List[Tensors] = []
 
-        for tensor in input:
+        # Handle sequences for backward compatibility.
+        if len(inputs) == 1 and isinstance(inputs[0], Sequence):
+            is_single_sequence = True
+            inputs = inputs[0]  # type: ignore[assignment]
+
+        for tensor in inputs:
             tensors = tensor.chunk(chunks)
-            rotated.append(cast(Tensors, tensors))
+            rotated.append(tensors)
 
-        inputs = zip(*rotated)
+        unwrapped_inputs = zip(*rotated)  # type: ignore[assignment]
 
-    return [Batch(x) for x in inputs]
+    return [Batch(x, is_single_sequence) for x in unwrapped_inputs]
```

Comment:
Suppose the user creates `Pipe` for `nn.Sequence(m1: M1, m2: M2)` with following signature:
```py
class M1(nn.Module):
   def forward(self, input: Tensor) ->  Tuple[Tensor, Tensor]:
        ...
class M2(nn.Module):
   def forward(self, input: Tuple[Tensor, Tensor]) ->  Tensor:
        ...
```
This works fine with old implementation, but it seems to me with new implementation we set is_single_sequence=False, which is wrong.

Reply:
@mrzzd I thought about this for a while and couldn't come up with a good solution. I did consider your proposal about adding `is_single_sequence` to the constructor. The tricky thing with that is that if by default that is False, we anyways end up breaking compatibility since the user has to manually set this flag to get the old behavior.

I felt it was probably better to just break compatibility since the feature is Beta and keep the behavior simple and consistent without special handling for tuple of tensors. We can call this out in our release notes as well for the next release. As a result, I removed most of the logic pertaining to is_single_sequence.

cc @mrshenli